### PR TITLE
hasCenterLegend to hide/show center legend

### PR DESCRIPTION
**Pull request by rajeshmr82**
fix - Lengthy names are breaking the Donut Chart and the legend

## Description

Introduces a new attribute hasCenterLegend  which is used to show/hide center legend

## Motivation and Context

hasCenterLegend  can be used to show/hide center legend in case the length of legend is too long. It can be complemented with displaying the whole text inside a mini-tooltip

## How Has This Been Tested?

Additional unit test has been added. Tested locally along with regression testing

## Types of changes

-   [ ] Refactor (changes the way we code something without changing its functionality)
-   [ ] Bug fix (non-breaking change which fixes an issue)
-   [x] New feature (non-breaking change which adds functionality)
-   [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

-   [ ] Latest master code has been merged into this branch
-   [ ] No commented out code (if required, place // TODO above with explanation)
-   [ ] No linting issues
-   [ ] Build is successful
-   [ ] Code follows the [API Guidelines](http://britecharts.github.io/britecharts/topics-index.html#toc5__anchor)
-   [ ] Updated the documentation
-   [ ] Added tests to cover changes
-   [ ] All new and existing tests passed

### CODE REVIEW

File: src/charts/line/line.js [hunk 1]
```diff
@@ -696,23 +698,19 @@ export default function module() {
...
+                    'stroke-dasharray',
+                    totalLength + ' ' + strokeDasharrayOffset * totalLength
+                )
+                .attr('stroke-dashoffset', totalLength)
                 .transition()
                 .duration(animationDuration)
                 .ease(ease)
-                .attr('x', width)
-                .on('end', () => maskingRectangle.remove());
+                .attr('stroke-dashoffset', strokeDashoffset);
```
[the remark this answers is not in the document]
Comment (rajeshmr82) [reply src/charts/line/line.js]: @Golodhros Thanks for the review. I have addressed all the issues

File: src/charts/donut/donut.js [hunk 2]
```diff
@@ -994,5 +995,22 @@ export default function module() {
...
 
+    /**
+     * Gets or Sets the hasCenterLegend property of the chart, making it display
+     * legend at the center of the donut.
+     *
+     * @param  {boolean} _x         If we want to show legent at the center of the donut
+     * @return {boolean | Module}   Current hasCenterLegend flag or Chart module
+     * @public
+     */
+    exports.hasCenterLegend = function (_x) {
```
[the remark this answers is not in the document]
Comment (rajeshmr82) [reply src/charts/donut/donut.js]: Reordered accessors to be in alphabetic order

File: src/charts/donut/donut.spec.js [hunk 3]
```diff
@@ -641,6 +641,18 @@ donutDataSets.forEach((datasetName) => {
                 expect(previous).not.toBe(expected);
                 expect(actual).toBe(expected);
             });
+
+            it('should provide hasCenterLegend getter and setter', () => {
```
[the remark this answers is not in the document]
Comment (rajeshmr82) [reply src/charts/donut/donut.spec.js]: Reordered test cases to be in alphabetic order similar to accessors

**Review by Golodhros — APPROVED** [review]
Thanks a lot @rajeshmr82!

**Pull request merged**